lock/common.go

```diff
@@ -0,0 +1,32 @@
+// Package lock provides implementations for cancellable and distributed locks.
+package lock
+
+import (
+	"golang.org/x/net/context"
+	"gopkg.in/src-d/go-errors.v0"
+)
+
+var (
+	ErrCanceled      = errors.NewKind("context canceled")
+	ErrAlreadyClosed = errors.NewKind("already closed")
+)
+
+// Locker is a service providing locks.
+type Locker interface {
+	// Lock acquires a lock with the given id.
+	// Context can be used to cancel the operation (e.g. with a timeout).
+	// If it succeeds, it returns a Lock. Otherwise, it returns an error.
+	Lock(id string, ctx context.Context) (Lock, error)
```

Comment:
Wich error in each case? Per example, witch error will I receive if the lock is already taken? or if I lost the connection with the locking provider?